components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/impl/OrganizationManagementDAOImpl.java

```diff
@@ -1321,6 +1321,32 @@ public void addRootOrganization(Organization rootOrganization) throws Organizati
         }
     }
 
+    @Override
+    public List<String> getOrganizationsMetaAttributes(boolean recursive, Integer limit, String organizationId,
+                                                       String sortOrder, List<ExpressionNode> expressionNodes,
+                                                       List<ExpressionNode> parentIdExpressionNodes)
+            throws OrganizationManagementServerException {
+
+        FilterQueryBuilder filterQueryBuilder = buildFilterQuery(expressionNodes);
+        FilterQueryBuilder parentIdFilterQueryBuilder = buildParentIdFilterQuery(parentIdExpressionNodes);
+
+        String sqlStmt = appendFilterQueries(GET_ORGANIZATIONS_META_ATTRIBUTES, filterQueryBuilder,
+                parentIdFilterQueryBuilder, GET_ORGANIZATIONS_META_ATTRIBUTES_TAIL,
+                recursive, sortOrder);
+
+        List<String> organizationMetaAttributes;
+        NamedJdbcTemplate namedJdbcTemplate = Utils.getNewTemplate();
+        try {
+            organizationMetaAttributes = namedJdbcTemplate.executeQuery(sqlStmt,
+                    (resultSet, rowNumber) -> resultSet.getString(1),
+                    namedPreparedStatement -> setPreparedStatementParams(namedPreparedStatement, organizationId,
+                            null, limit, filterQueryBuilder, parentIdFilterQueryBuilder, null));
+        } catch (DataAccessException e) {
+            throw handleServerException(ERROR_CODE_ERROR_RETRIEVING_ORGANIZATIONS, e);
```

Comment:
This should return a new error related to orgs' attribute retrieving failure

Reply:
A new error is now added as `"ERROR_CODE_ERROR_RETRIEVING_ORGANIZATIONS_META_ATTRIBUTES".`